docs/advanced_documentation/build-guide.md

```diff
@@ -110,6 +110,13 @@ Then you can run the tests.
 pytest
 ```
 
+A basic `self_test` function is provided to check if the installation was successful and ensures there are no build errors, segmentation faults, undefined symbols, etc. It performs multiple C API calls, runs through the main data flow, and verifies the integrity of serialization and deserialization.
```

Comment:
and yes i know that i usually do this incorrectly as well and that's OK as long as it doesn't end up in production 😬 

```suggestion
A basic `self_test` function is provided to check if the installation was successful and ensures there are no build errors, segmentation violations, undefined symbols, etc. It performs multiple C API calls, runs through the main data flow, and verifies the integrity of serialization and deserialization.
```